core/src/main/java/io/seata/core/store/db/LogStoreDataBaseDAO.java

```diff
@@ -280,7 +281,12 @@ public boolean updateGlobalTransactionDO(GlobalTransactionDO globalTransactionDO
             ps = conn.prepareStatement(sql);
             ps.setInt(1, globalTransactionDO.getStatus());
             ps.setString(2, globalTransactionDO.getXid());
-            return ps.executeUpdate() > 0;
+            ps.setInt(3, globalTransactionDO.getVersion());
+            int cnt = ps.executeUpdate();
+            if(cnt != 1) {
+                throw new StoreException("GlobalSession was updated or deleted by another transaction");
```

Comment:
no retry?

Reply:
In my opinion, retry is not necessary.

Reply:
I think, ideally, throwing this exception is a small probability event. If failed, retry will triggered by high level code at a later time.